cfme/infrastructure/virtual_machines.py

```diff
@@ -931,35 +931,25 @@ def reconfigure(self, new_configuration=None, changes=None, cancel=False):
                 else:
                     mode = disk.mode
                     dependent = True
+
                 row = vm_recfg.disks_table.click_add_disk()
                 row.type.fill(disk.type)
-                # Unit first, then size (otherwise JS would try to recalculate the size...)
-                from cfme.infrastructure.provider.rhevm import RHEVMProvider
-                if self.provider.one_of(RHEVMProvider):
-                    # TODO: Workaround necessary until BZ 1524960 is resolved
-                    # -- block start --
-                    row[3].fill(disk.size_unit)
-                else:
-                    if self.appliance.version < '5.9':
-                        unit_column = 4
-                    else:
-                        unit_column = 5
-                    # -- block end --
-                    row[unit_column].fill(disk.size_unit)
-                    row.mode.fill(mode)
-                    row.dependent.fill(dependent)
+                row.mode.fill(mode)
                 row.size.fill(disk.size)
+                row.unit.fill(disk.size_unit)
+                row.dependent.fill(dependent)
                 row.actions.widget.click()
                 disk_message = 'Add Disks'
             elif action == 'delete':
                 row = vm_recfg.disks_table.row(name=disk.filename)
                 # `delete_backing` removes disk from the env
                 row.delete_backing.fill(True)
-                row.actions.widget.click()
+                row[9].widget.click()
```

Comment:
@mshriver can you think of something better here? I feel `row[9]` is very not descriptive. Maybe add some comments here?

Reply:
@jawatts agreed, adding